### ISSUE TITLE
triangulate_points(solver="cofactor") returns NaN for float16 pixel input: null vector downcast before normalising

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4863]
`triangulate_points(..., solver="cofactor")` casts float16/bfloat16 input up to float32 and solves two `3 x 4` sub-systems with `null_vector_3x4`. It then casts the two null vectors **back to the input dtype** (`null_vector_3x4(A_012).to(row0.dtype)`) and only after that normalises them. For pixel coordinates and a `K` with focal lengths in the hundreds, the cofactor null vector reaches about `2e8`, far above the float16 maximum of `65504`. The downcast turns it into `inf`, the normalisation `inf / inf` turns it into `NaN`, and every output row is `NaN`. `svd` and `eigh` on the same float16 input are finite and accurate to `0.0078`, and `cofactor` on bfloat16 is finite because bfloat16 has float32's exponent range. A user would expect the three solvers to differ only in accuracy and speed, as the docstring describes them, and not for one of them to fail on ordinary input.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import projection_from_KRt, triangulate_points
from kornia.geometry.solvers import null_vector_3x4

K1 = torch.tensor([[800.0, 0.0, 320.0], [0.0, 760.0, 200.0], [0.0, 0.0, 1.0]])
K2 = torch.tensor([[700.0, 0.0, 300.0], [0.0, 740.0, 240.0], [0.0, 0.0, 1.0]])
R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]]))[0]
t = torch.tensor([[0.5], [0.1], [0.02]])
P1 = projection_from_KRt(K1, torch.eye(3), torch.zeros(3, 1))
P2 = projection_from_KRt(K2, R, t)
X = torch.tensor([[0.2, -0.1, 4.0], [-0.3, 0.25, 5.0], [0.1, 0.3, 4.5], [-0.5, -0.4, 6.0]])


def project(P, X):
    x = torch.cat([X, torch.ones(len(X), 1)], 1) @ P.T
    return x[:, :2] / x[:, 2:]


x1, x2 = project(P1, X), project(P2, X)
for dtype in [torch.float32, torch.float16, torch.bfloat16]:
    for solver in ["svd", "eigh", "cofactor"]:
        Xt = triangulate_points(P1.to(dtype), P2.to(dtype), x1.to(dtype), x2.to(dtype), solver=solver)
        err = (Xt.float() - X).abs().max().item()
        print(f"{str(dtype):15s} {solver:8s} NaN rows {int(Xt.isnan().any(1).sum())}/{len(X)}  max|X - X_true| {err:.3g}")

# the cofactor null vector, computed in float32, is cast back to float16 before it is normalised
h = P1.half()
A = torch.stack([x1.half()[:, 0:1] * h[2:3] - h[0:1], x1.half()[:, 1:2] * h[2:3] - h[1:2],
                 x2.half()[:, 0:1] * P2.half()[2:3] - P2.half()[0:1]], -2)
v = null_vector_3x4(A.float())
print("max |null vector| in float32:", v.abs().max().item(), "| float16 max:", torch.finfo(torch.float16).max,
      "| inf after .half():", bool(v.half().isinf().any()))
```

```
torch.float32   svd      NaN rows 0/4  max|X - X_true| 4.77e-07
torch.float32   eigh     NaN rows 0/4  max|X - X_true| 4.77e-07
torch.float32   cofactor NaN rows 0/4  max|X - X_true| 2.38e-06
torch.float16   svd      NaN rows 0/4  max|X - X_true| 0.00781
torch.float16   eigh     NaN rows 0/4  max|X - X_true| 0.00781
torch.float16   cofactor NaN rows 4/4  max|X - X_true| nan
torch.bfloat16  svd      NaN rows 0/4  max|X - X_true| 0.125
torch.bfloat16  eigh     NaN rows 0/4  max|X - X_true| 0.125
torch.bfloat16  cofactor NaN rows 0/4  max|X - X_true| 0.0938
max |null vector| in float32: 214472000.0 | float16 max: 65504.0 | inf after .half(): True
```

kornia main 7ddf731bd, torch 2.14.0 (and 2.5.1: same), CPU float16, macOS arm64.

## Classification

Clearly broken output. No idiom of 1–9; candidate: "downcast before normalising". The upcast to a wider compute dtype is undone before the step that brings the value back into range, so an intermediate overflows in the output dtype. It is related to Idiom 2 (dtype handling) but is a different failure.

## Disposition

Focused fix, welcome as a PR: normalise `h_012` and `h_123` (and align their signs and sum them) in the compute dtype, and cast to the input dtype only once, on the final homogeneous vector `points3d_h`, whose entries then have magnitude at most `2`. Checked out of tree by re-running the cofactor branch with that order on the repro input: float16 output has no `NaN` and `max|X - X_true| = 0.0078125`, the same as `svd` and `eigh`. float32 and float64 results are bitwise unchanged, because the cast is a no-op there. bfloat16 moves by at most one bfloat16 ulp (`0.03125` here), and its error to `X_true` stays `0.09375`.
Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4865 (same function: a point at infinity comes back as a finite point).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
